### DISCUSSION TITLE
Can't install on Big Sur (Intel MBP)

### DISCUSSION THREAD

**Discussion by mweintr** [Homebrew/discussions#140] [Everyday usage]
Fresh install of Big Sur (11.0.1). I ran `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"` and get this output:

> % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
> ...
> Press RETURN to continue or any other key to abort
> ==> Downloading and installing Homebrew...
> /usr/local/homebrew/.git: Permission denied
> Failed during: git init -q

**Comment by SMillerDev** [comment]
You don't have permission to write to `/usr/local/homebrew/.git` as it says. Make sure you have write permission there.

**Reply by mweintr** [reply]
Thank you. This was a fresh install of Big Sur, so I didn't think that was necessary as the docs didn't mention that.

**Reply by glessard** [reply]
The script should work on a freshly-installed system. It does not.

**Reply by SMillerDev** [reply]
I've never seen this issue setting up Homebrew on fresh installs. Are you sure your `/usr/local` isn't locked by something like Jamf?

**Reply by mweintr** [reply]
In my case, `/usr/local` was not locked by Jamf, as I don't have Jamf controlling that Mac.

**Reply by SMillerDev** [reply]
Do you have something else controlling your Mac? Potentially a virusscanner or other management software?

**Reply by mweintr** [reply]
Nothing of the sort. This was a fresh installation of Big Sur on a MBP (2015). I think this was one of the first things I installed on this Mac, along with 1Password.

**Reply by gromgit** [reply]
And you're sure that Homebrew was never installed before? If this were a fresh install, and the installer had created brand-new directories:
> ```
> /usr/local/homebrew/.git: Permission denied
> ```
`homebrew` should've been `Homebrew` in that error.

**Reply by mweintr** [reply]
Yes, while Homebrew was installed in the previous installation (of whatever version of macOS preceded High Sierra), this was a fresh installation of HS on the MacBook in question.

**Reply by gromgit** [reply]
Unless you erased the partition containing `/usr/local` before installing Big Sur, it's almost certain that you're actually looking at your old Homebrew directory there, hence the bizarre error.

At this point, you might want to `sudo rm -fr /usr/local/homebrew`, then run the installer again.

**Reply by mweintr** [reply]
I wiped out all partitions before installing Big Sur. 

Regardless, I don't need help anymore, as I manually changed the permissions on that `/usr/local/homebrew`. I'm just continuing this conversation to give you more information to prevent this happening to others.

**Reply by glessard** [reply]
Sorry for not replying earlier... in my case I was installing to a new partition, and I used Migration Assistant to copy Applications (only) to my new partition before continuing. I have since found many things copied to unexpected places, and the only possible cause is MA. I don't know for sure what happened with /usr/local/homebrew since after commenting here I `rm -rf`ed my `/usr/local` -- and after that things worked correctly. I've concluded that Migration Assistant should not be used for anything beyond user accounts in order to avoid surprises.

This being said, there could have been a more useful failure message, rather than just relaying git's output.

**Reply by PlaidRadish** [reply]
Same, here, on fresh install of macOS Big Sur. 
Starts with "error: could not lock config file /usr/local/Homebrew/.git/config: Permission denied"
Followed by "fatal: could not set 'core.repositoryformatversion' to '0'"
Then, "Failed during: git init -q"

Is my issue the same, or different?

**Reply by glessard** [reply]
Looks the same to me. Try removing all of /usr/local/Homebrew and start over?

**Reply by tlk** [reply]
@PlaidRadish Did you use the Migration Assistant? Could you share the output of `ls -la  /usr/local/Homebrew/` and `ls -la /usr/local/Homebrew/.git`?

**Reply by PlaidRadish** [reply]
@glessard - indeed, removing everything from /usr/local/ folder allowed me to start homebrew installation over again. Unfortunately, the downloads were incredibly slow.

@tlk - sorry, I went ahead and tried the suggested fix, so I guess it's too late for that...

**Reply by Linus-XZX** [reply]
My personal understanding: `/usr/local` itself is now 755 `root:wheel` starting macOS 11, so a solution can be `sudo chown -Rv $(whoami):staff /usr/local/Homebrew` (I included -v for safety, you can check the output in case things got wrong). Before macOS 11 `/usr/local` and everything in it is actually `$(whoami):staff`

This should also apply if you use Homebrew Cask later on, because it uses `/usr/local/Caskroom` which works similarly. After installing Homebrew this will also happen to `/usr/local/Cellar`.

Edit: For a fresh install this problem will be moot because install.sh uses `sudo` to create these three directories and set the proper access rights.

**Reply by PlaidRadish** [reply]
> My personal understanding: `/usr/local` itself is now 755 ... 
> This should also apply if you use Homebrew Cask later on, because it uses `/usr/local/Caskroom` ...

So, have the Homebrew devs added this to the installer script (recognizing Big Sur and later) so that it automates during install?

**Reply by Linus-XZX** [reply]
> > My personal understanding: `/usr/local` itself is now 755 ...
> > This should also apply if you use Homebrew Cask later on, because it uses `/usr/local/Caskroom` ...
> 
> So, have the Homebrew devs added this to the installer script (recognizing Big Sur and later) so that it automates during install?

Didn't see it on the script as of yesterday... My take is that there was a somewhat broken installation of Homebrew with a 755 `root:wheel /usr/local/homebrew`, so the installer skipped the `mkdir` step that uses `sudo` to deal with privilege issues.

**Reply by gromgit** [reply]
https://github.com/Homebrew/install/pull/528 has been merged to force Homebrew ownership on all files and directories under existing Homebrew-specific hierarchies (including `/usr/local/Homebrew`). Running the installer again should Do The Right Thing.

**Reply by PlaidRadish** [reply]
Homebrew devs rock.
